content/protobuf/language-mapping/service.md

```diff
@@ -0,0 +1,220 @@
+while a server application hosts a service that implements this service and all its methods.
+
+For example:
+
+```protobuf
+syntax = "proto3";
+
+package visitor_center;
+option csharp_namespace = "VisitorCenter";
+
+// A simple Greeter service.
+service Greeter {
+    // Creates a personalized greeting.
+    rpc Greet (GreetRequest) returns (GreetResponse);
+}
+
+// The request contains the name of the person to greet.
+message GreetRequest {
+    string name = 1;
+}
+
+// The response contains the greeting.
+message GreetResponse {
+    string greeting = 1;
+}
+```
+
+All RPC methods are abstract at the Protobuf level: you can't implement a method in the Protobuf language.
+
+In order to implement a Protobuf service, you need a RPC framework that provides support for Protobuf. We describe here
+IceRPC's support for Protobuf.
+
+## C# mapping
+
+The `protoc-gen-icerpc-csharp` plugin compiles Protobuf service _Name_ into two C# interfaces (I*Name* and
+I*Name*Service) and one C# struct (*Name*Client). The identifiers of the generated interfaces and struct are always in
+Pascal case, per the usual C# naming conventions, even when _Name_ is not in Pascal case.
+
+### I*Name*
+
+Interface I*Name* provides the client-side API that allows you to call the remote IceRPC service that implements the
+the associated Protobuf service. It's a minimal interface with an abstract method for each RPC method defined in this
+Protobuf service.
+
+For example:
+
+{% side-by-side alignment="top" %}
+
+```protobuf
+syntax = "proto3";
+package example;
+option csharp_namespace = "Example";
+
+service Widget {
+    rpc Spin (SpinConfig)
+        returns (google.protobuf.Empty)
+}
+```
+
+```csharp
+namespace Example;
+
+public partial interface IWidget
+{
+    // One C# method per rpc method
+    Task<WellKnownTypes.Empty> SpinAsync(
+        SpinConfig message,
+        IFeatureCollection? features = null,
+        CancellationToken cancellationToken =
+            default);
+}
+```
+
+{% /side-by-side %}
+
+### *Name*Client
+
+The generated record struct *Name*Client implements I*Name* by sending requests to a remote service with IceRPC.
+
+In order to call a remote service, you need to construct a client struct using one of its "invoker" constructors:
+
+```csharp
+public readonly partial record struct WidgetProxy : IWidget
+{
+    public WidgetProxy(
+        IInvoker invoker,
+        ServiceAddress? serviceAddress = null,
+        ProtobufEncodeOptions? encodeOptions = null)
+    {
+        ...
+    }
+
+    public WidgetProxy(
+        IInvoker invoker,
+        System.Uri serviceAddressUri,
+        ProtobufEncodeOptions? encodeOptions = null)
+        : this(invoker, new ServiceAddress(serviceAddressUri), encodeOptions)
+    {
+    }
+}
+```
+
+The `invoker` parameter represents your [invocation pipeline](/icerpc/invocation/invocation-pipeline), the
+`serviceAddress` or `serviceAddressUri` parameter corresponds to the
```

Comment:
Even if you'd only ever use one at a time, we are still talking about 2 parameters, plural.
```suggestion
`serviceAddress` and `serviceAddressUri` parameters correspond to the
```